es-ds-components/components/es-toggle.vue

```diff
@@ -0,0 +1,133 @@
+<script setup lang="ts">
+import { SwitchRoot, SwitchThumb } from 'reka-ui';
+
+const switchId = useId();
+const model = defineModel<boolean>({
+    default: false,
+});
+
+defineProps({
+    disabled: {
+        type: Boolean,
+        default: false,
+        description: 'Disables the toggle.',
+    },
+});
+
+// Workaround: Reka UI components don't play nice with the Nuxt 3.16+ dev server side rendering while using NPM link
+const wrappingComponent = import.meta.dev ? resolveComponent('ClientOnly') : 'span';
```

Comment:
The fallback element here might need to be a `'div'` rather than a `'span'` because I think it's technically invalid HTML to have a `<div>` (block element) inside of a `<span>` (inline element).